### PR TITLE
 [WIP] Wallet: Cache CWalletDB pointer in CWallet to improve performance

### PR DESCRIPTION
**Pull request by pstratem**

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Can you fix the tests for this?

**Comment by pstratem** [comment]
@TheBlueMatt I'm not sure whether it's broken or if the tests are broken.

**Comment by pstratem** [comment]
Indeed all the tests pass except for the tests which fail on master #8045

**Comment by MarcoFalke** [comment]
@pstratem The `backupwallet` rpc appears to be no longer working

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -605,12 +605,27 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
...
+    }
+
+    void OpenDB()
+    {
+        pwalletdb = new CWalletDB(strWalletFile, "cr+");
+    }
+
+    void CloseDB()
+    {
+        if (pwalletdb) {
```
Comment (sipa) [inline src/wallet/wallet.h]: The conditional is unneeded: deleting NULL is defined to be a nop.

**Comment by pstratem** [comment]
Going to reorganize these changes

**Pull request closed without merging**